### ISSUE TITLE
Update Puppeteer

### ISSUE THREAD

**Issue by nfmohit** [google/site-kit-wp#5862]
## Feature Description

While working with #5154, we noticed that the Puppeteer version being used is quite outdated. As a result, we're not able to use new methods like `waitForNetworkIdle` which were introduced in the newer versions. We should try to update Puppeteer (and preferably its related packages) to the latest version without breaking anything.       

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The puppeteer dependency is upgrade to its latest version if possible. If impossible, then it should be updated to a version that we can without rewriting a lot of tests.

## Implementation Brief

* [ ] Once #10013 is merged, update the packages `jest-puppeteer` and `puppeteer` to the latest version.
* [ ] Address any issues causing E2E tests to break:
   * TODO: IB author should check issues that this causes within the E2E test suite to assess work needed to fix in a draft PR.

**Related fix**
* [ ] Remove the console ignore for blockEditor.transformStyles in `tests/e2e/config/bootstrap.js` (details in #11117)

### Test Coverage

* E2E tests should be passing reliably in develop at the point of working on this issue (Some known issues are tracked through tickets such as https://github.com/google/site-kit-wp/issues/8229), ensure that no new E2E failures or instability introduced in this issue.

## QA Brief

* 

## Changelog entry

*

**Comment by eugene-manuilov** [comment]
@tofumatt @aaemnnosttv could you please check AC and notes in IB and give me your thumb up if it looks good to you?

**Comment by aaemnnosttv** [comment]
@eugene-manuilov SGTM, let's give it a try 👍

**Comment by benbowler** [comment]
After testing this it's definitely not simple to jump straight to the latest version of puppeteer. However splitting the E2E infrastructure unlocks the upgrade pathway.

I suggest we take the approach of #6357, by creating tickets to upgrade to the next version only, and once finishing each version upgrade, opening a new ticket to work on the next major version upgrade.

**Comment by eugene-manuilov** [comment]
I think this ticket will be addressed when we work on the #10013. Perhaps we should close this one or priorities that one to complete first and then decide whether to close this one.

**Comment by benbowler** [comment]
Hey @eugene-manuilov, based on the comment on the IB, the test PR #10020 does successfully split the E2E package to it's own package.json and updates the E2E workflow to be able to switch node versions and install it's own packages and run tests in this location.

If we close this issue I suggest linking the PR #10020 in a comment or the description as this could shortcut some of the requirements in that ticket.

Also in the experimentation in this ticket I found that with a separate package.json we can upgrade the packages as follows without all E2E tests failing:

```
"jest-puppeteer": "^6.2.0",
"puppeteer": "^14.4.1"
```

So we could do that there then create work as we are doing in https://github.com/google/site-kit-wp/issues/6357 by creating a string of follow up issues "Upgrade puppeteer to the next major version".

**Comment by eugene-manuilov** [comment]
@benbowler 

> Also in the experimentation in this ticket I found that with a separate package.json we can upgrade the packages as follows without all E2E tests failing:
> 
> ```
> "jest-puppeteer": "^6.2.0",
> "puppeteer": "^11.0.0"
> ```

Do you remember why you couldn’t upgrade it to the latest version? I think we need to aim upgrading it to the latest version in one go rather than fixing issues with intermediate versions.

> _Some thoughts to IB author:_
> 
> _Since we use puppeteer as a direct dependency for e2e tests only, we can probably decouple e2e related scripts and dependencies and put them within e2e tests folder as a separate package.json file. In this case, these dependencies won't conflict with the dependencies that we need for development and storybook stories and upgrade can go smoother. Of course, this will require updating e2e workflows to account for the new location of e2e node modules._

This is not needed in IB, it was for you and wasn't supposed to appear in the final IB. Please, remove it.

> If we close this issue I suggest linking the PR 10020 in a comment or the description as this could shortcut some of the requirements in that ticket.

No need to close this one, let's keep it to fix the problem.

**Comment by ivonac4** [comment]
Hey @benbowler , still planning to work on this one? If not, could you please unassign your self so someone from Team M can pick it up? Thank you

**Comment by benbowler** [comment]
@ivonac4 yes. It's waiting on the node upgrade ticket as many of these upgrade tickets are. Hence why I've not updated it yet.

**Comment by ivonac4** [comment]
Okay, cool. Will push for that one to be done in the next sprint, hopefully.

**Comment by benbowler** [comment]
@ankitrox you can take this over if you like.

**Comment by binnieshah** [comment]
@abdelmalekkkkk should we reassign this one?

**Comment by techanvil** [comment]
As discussed on [Slack](https://fueled.slack.com/archives/C077DB69US2/p1767782737992749), we're closing this issue and will proceed with the migration to Playwright instead.
